File: annotations/src/main/java/com/google/j2objc/NotImportedError.java
package com.google.j2objc;

import com.google.j2objc.NotImportedError;

public class NotImportedError extends RuntimeException {
	
	public NotImportedError(Object... objects) {
		throw this;
	}
	
	public static Object throwUnreachableObjectError(Object... objects) {
		throw new NotImportedError();
	}

	public static byte throwUnreachablePrimitiveError(Object... objects) {
		throw new NotImportedError();
	}
}

File: jre_emul/Classes/ARGC/NotImportedError.m
#import "J2ObjC_common.h"

#if J2OBJC_USE_GC
#include "com/google/j2objc/NotImportedError.h"

@implementation ComGoogleJ2objcNotImportedError
- (instancetype)initWithNSObjectArray:(IOSObjectArray *)objects {
  [NSException raise:@"NSException" format:@"Unreachabe access error"];
  return nil;
}

+ (id)throwUnreachableObjectErrorWithNSObjectArray:(IOSObjectArray *)objects {
  [NSException raise:@"NSException" format:@"Unreachabe access error"];
  return nil;
}

+ (jbyte)throwUnreachablePrimitiveErrorWithNSObjectArray:(IOSObjectArray *)objects {
  [NSException raise:@"NSException" format:@"Unreachabe access error"];
  return 0;
}

@end

id new_ComGoogleJ2objcNotImportedError_initWithNSObjectArray_(IOSObjectArray * objects) J2OBJC_METHOD_ATTR NS_RETURNS_RETAINED
{
  [NSException raise:@"NSException" format:@"Unreachabe access error"];
  return nil;
}

id ComGoogleJ2objcNotImportedError_throwUnreachableObjectErrorWithNSObjectArray_(IOSObjectArray * objects) {
  [NSException raise:@"NSException" format:@"Unreachabe access error"];
  return nil;
}

#endif
